### ISSUE TITLE
Request: sp-barf-hybrid-sexp

### ISSUE THREAD

**Issue by Wasabichicken** [Fuco1/smartparens#570]
For C/C++ modes, `sp-slurp-hybrid-sexp` is a godsend. It solves several issues I've got with the regular `sp-forward-slurp-sexp`, e.g. weird handling of the C++ arrow (->) operator and strange insertion of whitespace (see https://www.reddit.com/r/emacs/comments/451wn7/smartparens_for_clike_modes/)

However, there doesn't seem to be a `sp-barf-hybrid-sexp` available that undoes what `sp-slurp-hybrid-sexp` does. Since I suck at lisp and the SP source looks strange to me, consider this a humble feature request.

**Comment by Wilfred** [comment]
Agreed, there's no way of doing this currently and it would be nice.

**Comment by Wilfred** [comment]
FWIW, I explicitly fixed the whitespace issue, but only for Python, in #561. I don't think it's worth doing in general though -- for non-lisps the hybrid-sexp commands are better.

**Comment by Fuco1** [comment]
> and the SP source looks strange to me

don't worry, it's like that for everyone.
